diff --git a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
index 623d90c..90d6362 100644
--- a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
+++ b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
@@ -2591,9 +2591,11 @@ public class ReactExoplayerView extends FrameLayout implements
         } else {
             rootView.removeView(exoPlayerView);
             if (!rootViewChildrenOriginalVisibility.isEmpty()) {
-                for (int i = 0; i < rootView.getChildCount(); i++) {
+                int visibilityCount = rootViewChildrenOriginalVisibility.size();
+                for (int i = 0; i < rootView.getChildCount() && i < visibilityCount; i++) {
                     rootView.getChildAt(i).setVisibility(rootViewChildrenOriginalVisibility.get(i));
                 }
+                rootViewChildrenOriginalVisibility.clear();
                 addView(exoPlayerView, 0, layoutParams);
                 reLayoutControls();
             }
@@ -2621,8 +2623,9 @@ public class ReactExoplayerView extends FrameLayout implements
         ViewGroup rootView = decorView.findViewById(android.R.id.content);
 
         if (!rootViewChildrenOriginalVisibility.isEmpty()) {
-            if (exoPlayerView.getParent().equals(rootView)) rootView.removeView(exoPlayerView);
-            for (int i = 0; i < rootView.getChildCount(); i++) {
+            if (rootView.equals(exoPlayerView.getParent())) rootView.removeView(exoPlayerView);
+            int visibilityCount = rootViewChildrenOriginalVisibility.size();
+            for (int i = 0; i < rootView.getChildCount() && i < visibilityCount; i++) {
                 rootView.getChildAt(i).setVisibility(rootViewChildrenOriginalVisibility.get(i));
             }
             rootViewChildrenOriginalVisibility.clear();
