diff --git a/node_modules/react-native-vision-camera-barcode-scanner/android/build.gradle b/node_modules/react-native-vision-camera-barcode-scanner/android/build.gradle
index c4dfdc5..a84041a 100644
--- a/node_modules/react-native-vision-camera-barcode-scanner/android/build.gradle
+++ b/node_modules/react-native-vision-camera-barcode-scanner/android/build.gradle
@@ -68,6 +68,7 @@ android {
   externalNativeBuild {
     cmake {
       path "CMakeLists.txt"
+      buildStagingDirectory file("${rootProject.buildDir}/short-cxx/${project.name}")
     }
   }
 
